features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -653,13 +653,13 @@ ble_error_t GenericSecurityManager::generateOOB(
     /* Secure connections. Avoid generating if we're already waiting for it.
      * If a local random is set to 0 it means we're already calculating. */
     if (!is_all_zeros(_oob_local_random)) {
-        status = _pal.generate_secure_connections_oob();
+        _oob_local_address = *address;
+        /* this will be updated when calculation completes,
+         * a value of all zeros is an invalid random value */
+        set_all_zeros(_oob_local_random);
```

Comment:
This modification  to `_oob_local_random` should be reverted if `generate_secure_connections_oob` fail.

Reply:
no problem, should be easy enough.